### ISSUE TITLE
minecraft:block_placer > replace_block_item validation

### ISSUE THREAD

**Issue by Xterionix** [Blockception/BC-Minecraft-Bedrock-Diagnoser#282]
The field requires that the block that is being placed matches the id of the item for the replace_block_item field to be valid. 

I'd do this myself, but after going through `BehaviorPack/Item/components/diagnose.ts` I'm not sure how to access the item's identifier from the component test aside from calling `diagnoser.document.getText()` and parsing the JSON again. Seems redundant to parse it again since it's already been done in the `entry.ts` file

**Comment by DaanV2** [comment]
Do you have a json example of what you mean? Hard time visualizing it atm

**Comment by Xterionix** [comment]
If this is the `item.json`
```json
{
  "format_version": "1.20.80",
  "minecraft:item": {
    "description": {
      "identifier": "test:block"
    },
    "components": {
      "minecraft:block_placer": {
        "block": "test:block",
        "replace_block_item": true
      }
    }
  }
}
```

`minecraft:block_placer.block` needs to equal `minecraft:item.description.identifier` if `replace_block_item` is true

**Comment by DaanV2** [comment]
Does it need to match? As per the documentation it just specifies a block. Do we need to check it equals the item, or the just an exiting block

https://learn.microsoft.com/en-us/minecraft/creator/reference/content/itemreference/examples/itemcomponents/minecraft_block_placer?view=minecraft-bedrock-stable

**Comment by Xterionix** [comment]
> Note: the identifier for this item must match the block's identifier for this field to be valid.
In reference to `replace_block_item` ^

It needs to match

**Comment by Xterionix** [comment]
On a similar note, for `minecraft:custom_components` format version needs to be >= 1.21.10 and I'd like to include a check for that as well, so I'd need access to the `item.json` from within the component test

**Comment by DaanV2** [comment]
I'll see if I can get some time on updating the framework, so you can easily add those checks

**Comment by Xterionix** [comment]
Great. Thank you

**Comment by Xterionix** [comment]
@DaanV2 Have you had a chance to work on this?

**Comment by DaanV2** [comment]
No, but now I am adding it: https://github.com/Blockception/BC-Minecraft-Bedrock-Diagnoser/pull/316
